routing/single_mwm_router.cpp

```diff
@@ -74,39 +81,37 @@ IRouter::ResultCode SingleMwmRouter::CalculateRoute(MwmSet::MwmId const & mwmId,
   }
 }
 
-IRouter::ResultCode SingleMwmRouter::DoCalculateRoute(MwmSet::MwmId const & mwmId,
-                                                      m2::PointD const & startPoint,
+IRouter::ResultCode SingleMwmRouter::DoCalculateRoute(m2::PointD const & startPoint,
                                                       m2::PointD const & /* startDirection */,
                                                       m2::PointD const & finalPoint,
                                                       RouterDelegate const & delegate,
                                                       Route & route)
 {
-  if (!mwmId.IsAlive())
-    return IRouter::RouteFileNotExist;
-
-  string const & country = mwmId.GetInfo()->GetCountryName();
+  // TODO: remove field m_country.
+  // Use m_countryFileFn(startPoint), m_countryFileFn(finalPoint) here.
+  auto const startCountry = platform::CountryFile(m_country);
+  auto const finishCountry = platform::CountryFile(m_country);
```

Comment:
Why do we need startCountry and finishCountry in SingleMwmRouter? The name of the router suggests that the country will be the same.